library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2464,7 +2464,7 @@ impl<T, A: Allocator> VecDeque<T, A> {
     ///
     /// let mut deque: VecDeque<_> = [0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55].into();
     /// let num = 42;
-    /// let idx = deque.partition_point(|&x| x < num);
+    /// let idx = deque.partition_point(|&x| x <= num);
     /// // The above is equivalent to `let idx = deque.binary_search(&num).unwrap_or_else(|x| x);`
```

Comment:
I haven't checked, but I imagine your chance makes it so that this comment is no longer accurate.

Reply:
Hi, thanks for the review, and thanks for spotting this.

The comment in `VecDeque` and `Vec` remain accurate if `num` is not found, which is the case here with `num` == 42.

The comment was not accurate to begin with when `num` is not unique, because `binary_search` is allowed to return any one of them.

```rs
use std::collections::VecDeque;

let mut deque: VecDeque<_> = [0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55].into();
let num = 1;
let idx = deque.partition_point(|&x| x < num);
assert_eq!(idx, 1);
let idx = deque.binary_search(&num).unwrap_or_else(|x| x);
assert_eq!(idx, 3);
let idx = deque.partition_point(|&x| x <= num);
assert_eq!(idx, 5);
deque.insert(idx, num);
assert_eq!(deque, &[0, 1, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55]);
```

You are right that the comment will become inaccurate when `num` is unique, since `partition_point` with `<=` would return the index after.

Using `<=` _will_ always help in the `Vec` case, and _may_ help in the `VecDeque` case (some cases may prefer `<`, but someone may copy that example for a regular `Vec`).

How about updating the comment to clarify that `binary_search` is equivalent to the `<` case if `num` is unique?